docs/copilot/guides/large-complex-codebases-guide.md

```diff
@@ -0,0 +1,408 @@
+
+Use `applyTo` patterns to limit AI context to relevant subsystems:
+
+- `.github/copilot-instructions.md` for repository-wide context
+- Focused `.instructions.md` files with specific `applyTo` patterns for subsystems
+- Reference documentation using Markdown links
+- Define module boundaries in module-specific instruction files
+
+Example scoping patterns:
+
+```markdown
+---
+applyTo: "src/api/**/*,tests/api/**/*"  # Backend API and tests only
+---
+
+---
+applyTo: "src/frontend/**/*.{ts,tsx}"   # Frontend TypeScript only
+---
+
+---
+applyTo: "**/*.{test,spec}.{js,ts}"     # Test files only
+---
+```
+
+## Advanced chat mode patterns
+
+### Role-based chat modes
+
+Different roles need different AI personas and tool access. Create [custom chat modes](/docs/copilot/customization/custom-chat-modes.md) with specific instructions and tool sets:
+
+```
+.github/chatmodes/
+  architect.chatmode.md      # System design, cross-cutting concerns
+  frontend-dev.chatmode.md   # UI components, user experience
+  backend-dev.chatmode.md    # APIs, data models, integrations
+  devops.chatmode.md         # Infrastructure, deployment, monitoring
+  security.chatmode.md       # Security review, threat modeling
+```
+
+Each chat mode file defines specific tools and instructions for that role's workflow.
+
+<!-- TODO: Add complete example chat mode files with tool configurations for each role -->
+
+### Team-specific workflows
+
+Create chat modes that encode team workflows:
+
+- **Research modes**: Read-only exploration using `search`, `codebase`, `usages` tools
+- **Migration modes**: Refactoring with specific instructions for technical debt reduction
+- **Integration modes**: Cross-service communication patterns with architectural context
+- **Review modes**: Code quality assessment with team guidelines
+
+### Cross-functional modes
+
+For work spanning multiple teams:
+
+- **API design mode**: Backend + frontend collaboration
+- **Performance mode**: Application + infrastructure optimization
+- **Migration mode**: Coordinated changes across multiple services
+
+## Context compaction techniques
+
+### Frequent intentional compaction
+
+In large codebases, context windows fill up quickly. Implement systematic context compaction:
+
+#### Memory file patterns
+
+Create dedicated files to preserve important context across chat sessions:
+
+```
+docs/ai/
+  current-tasks.md          # Active work and blockers
+  architectural-decisions.md # Key design choices and rationale
+  integration-patterns.md   # How services communicate
+  common-pitfalls.md        # Frequent mistakes and solutions
+```
+
+These files can be referenced in your chat mode instructions to provide persistent context.
+
+<!-- TODO: Add workflow for when and how to update memory files during development -->
+
+#### Chat mode delegation
+
+Use specialized chat modes for context-heavy tasks:
```

Comment:
By using specialized chat modes, how does this compact the context? Or does the user have to do something extra?